src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const stateHistory = [];
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const value of action.keysToRemove) {
+          delete newState[value];
+        }
+        break;
+
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
+        break;
+
+      default:
```

Comment:
Should add some error handling as a default case according to the task requirement
![9981](https://github.com/mate-academy/js_stateful-clones/assets/35075922/8e070594-8795-4e0b-8020-38d76cee98aa)